DataFormats/PatCandidates/interface/PackedGenParticle.h

```diff
@@ -100,121 +100,132 @@ namespace pat {
     }
 
     /// destructor
-    virtual ~PackedGenParticle();
+    ~PackedGenParticle() override;
     /// number of daughters
-    virtual size_t numberOfDaughters() const;
+    size_t numberOfDaughters() const override;
     /// return daughter at a given position (throws an exception)
-    virtual const reco::Candidate * daughter( size_type ) const;
+    const reco::Candidate * daughter( size_type ) const override;
     /// number of mothers
-    virtual size_t numberOfMothers() const;
+    size_t numberOfMothers() const override;
     /// return mother at a given position (throws an exception)
-    virtual const reco::Candidate * mother( size_type ) const;
+    const reco::Candidate * mother( size_type ) const override;
     /// direct access to the mother reference (may be null)
-    const reco::GenParticleRef & motherRef() const { return mother_; }
+    reco::GenParticleRef motherRef() const { 
+	if(mother_.isNonnull() && mother_.isAvailable()&& mother_->status()==1 ){ //if pointing to the pruned version of myself
+   	  if(mother_->numberOfMothers() > 0) 
+	  	  return mother_->motherRef(0); // return my mother's (that is actually myself) mother
+	 	else
+		  return edm::Ref<reco::GenParticleCollection>(); // return null ref
```

Comment:
with a dummy ```static const edm::Ref<reco::GenParticleCollection> nullGPR```
this method can retain it's "return by const ref" signature

Reply:
nope... the problem is in CompositePtrCandidate where motherRef(i) returns by value, not really in the nullRef(that I could handle the way you say)

Reply:
I meant CompositeRefCandidate... anyhow is in the call mother_->motherRef(0)